Source/engine.h

```diff
@@ -96,6 +97,25 @@ struct Point {
 		a -= b;
 		return a;
 	}
+
+	/**
+	 * @brief Computes fast approximate distance between this and other, using only integer arithmetic, with less than ~5% error
+	 * @param other Pointer to which we want the distance
+	 * @return Magnitude of vector this -> other
+	*/
+	int ApproxDistance(const Point &other) const

```

Comment:
nit: `Point` is tiny, pass it by value.

Reply:
Is there any overhead associated with passing it as a reference?

Reply:
Yes, sometimes. See the generated code: https://godbolt.org/z/4ojed9f1x

Reply:
Basically, a reference is effectively a pointer. So you're passing an address. The function needs to go to memory to fetch the values.

When you're passing a small value, it is passed in registers, the function has the values right away.